virttest/utils_test/libvirt.py

```diff
@@ -1333,17 +1333,25 @@ def do_migration(self, vms, srcuri, desturi, migration_type,
                                                           extra_opts))
                 migration_thread.start()
                 eclipse_time = 0
+                stime = int(time.time())
                 if func:
-                    stime = int(time.time())
-                    if func == process.run:
-                        try:
-                            func(args['func_params'], shell=args['shell'])
-                        except KeyError:
+                    # we execute command once the migration is started
+                    if self.check_migrate_started(vm):
+                        if func == process.run:
+                            logging.info("Migration started for %s", vm.name)
+                            try:
```

Comment:
maybe move try ... except to outer space to just under  check_migrate_started(vm)

Reply:
it is better to be in same place, as for process.run user may or may not give shell=True, it was meant to handle that.